### DISCUSSION TITLE
Next First Gejala (gejala pertama pada rule selanjutnya) pada halaman Admin

### DISCUSSION THREAD

**Discussion by 4njas** [rizkyilhampra/spdhtc#73] [Q&A] [answered]
halo bg, mau bertanya mengenai SPDHTC, untuk next first gejala itu maksudnya apa yah bg?, dan cara penggunaanya bagaimana?dihalaman admin bagian rule..

Terimakasih..
![Screenshot 2024-06-15 172736](https://github.com/rizkyilhampra/spdhtc/assets/86731218/b686e3a6-135d-4265-8cb6-e2de522992ee)

**Comment by rizkyilhampra** [comment, marked as answer]
Hallo. 

Aturan ini berkaitan dengan algoritma atau flow dari aplikasi ini. SPDHTC punya dua versi, seperti yang sudah dijelaskan [disini](https://github.com/rizkyilhampra/spdhtc?tab=readme-ov-file#sebelum-menjalankan-aplikasi-di-local) https://github.com/rizkyilhampra/spdhtc/blob/15a0ce9725473d6682703894bcd1e999ef15fd93/README.md?plain=1#L14-L18 

*Field* `next_gejala_id` hanya terdapat di versi aplikasi v2 ke atas, yang mana digunakan untuk 'lompat ke pertanyaan selanjutnya pada aturan selanjutnya'.
> Ini sebenarnya adalah solusi dari *case* kami yang mana membutuhkan untuk melompat ke aturan selanjutnya jika salah satu pertanyaan pada suatu aturan bernilai `false` atau 'tidak'.

Sebelumnya, untuk dataset aturan di SPDHTC bisa dilihat [disini](https://github.com/rizkyilhampra/spdhtc/blob/15a0ce9725473d6682703894bcd1e999ef15fd93/dataset.md) khususnya pada bagian **Aturan**

https://github.com/rizkyilhampra/spdhtc/blob/15a0ce9725473d6682703894bcd1e999ef15fd93/dataset.md?plain=1#L111-L120

Sehingga *field* `next_first_gejala_id` dapat diisi untuk 'gejala pertama' pada aturan selanjutnya. Misalkan aturan ke 1, punya 4 gejala yaitu G1, G2, G3, dan G4. Kemudian aturan ke 2, punya 4 gejala yaitu G5, G6, G7, dan G8. Maka untuk aturan ke 1 *field* `next_first_gejala_id` nya adalah G5.

Jika memilih versi 1.3, maka *field* ini tidak ada, namun algoritmanya juga berubah. Bisa dipertimbangkan untuk memakai algoritma yang mana menyesuaikan *case*-nya

**Reply by 4njas** [reply]
Halo,

Terima kasih banyak atas jawaban dan penjelasan mendetail yang Anda berikan mengenai algoritma/flow sistem pakar. Penjelasan Anda sangat membantu dan memberikan wawasan baru bagi saya dalam mengembangkan sistem ini. Kami sangat menghargai waktu dan usaha yang Anda luangkan untuk menjelaskan dengan rinci.

Jawaban Anda tidak hanya membantu saya memahami konsep ini dengan lebih baik, tetapi juga memberikan kami arahan yang jelas untuk implementasi lebih lanjut. Kami merasa sangat terbantu dengan adanya kontribusi dari komunitas yang luar biasa seperti Anda.

Sekali lagi, terima kasih atas bantuan dan dukungannya. Kami berharap bisa terus belajar dan berkolaborasi dengan Anda di masa mendatang.

Salam,
Anjasmara Hulu
